File: Dot examples/ASP Book Examples/Book Examples/Chapter17/CS/App_Code/Movie8.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Web.Configuration;
using System.Collections.Generic;

public class Movie8
{
    private static readonly string _connectionString;

    public DataTable GetAll()
    {
        // Initialize the DataAdapter
        SqlDataAdapter dad = new SqlDataAdapter("SELECT Title,Director FROM Movies", _connectionString);

        // Create a DataTable
        DataTable dtblMovies = new DataTable();

        // Populate the DataTable
        dad.Fill(dtblMovies);

        // Return results
        return dtblMovies;
    }

    static Movie8()
    {
        _connectionString = WebConfigurationManager.ConnectionStrings["Movies"].ConnectionString;
    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter17/VB/App_Code/Movie8.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Web.Configuration
Imports System.Collections.Generic

Public Class Movie8
    Private Shared ReadOnly _connectionString As String

    Public Function GetAll() As DataTable
        ' Initialize the DataAdapter
        Dim dad As New SqlDataAdapter("SELECT Title,Director FROM Movies", _connectionString)

        ' Create a DataTable
        Dim dtblMovies As New DataTable()

        ' Populate the DataTable
        dad.Fill(dtblMovies)

        ' Return results
        Return dtblMovies
    End Function

    Shared Sub New()
        _connectionString = WebConfigurationManager.ConnectionStrings("Movies").ConnectionString
    End Sub
End Class

